File: Brute_force.java
// 268. Missing Number
// Given the head of a singly linked list, group all the nodes with odd indices together followed by the nodes with even indices, and return the reordered list.
// The first node is considered odd, and the second node is even, and so on.
// Note that the relative order inside both the even and odd groups should remain as it was in the input.
// You must solve the problem in O(1) extra space complexity and O(n) time complexity.

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* oddEvenList(struct ListNode* head) {
    if(head==NULL || head->next==NULL)
        return head;
    struct ListNode *odd,*even,*evenHead;
    odd=head;
    even=head->next;
    evenHead=even;

    while(even!=NULL && even->next!=NULL)
    {
        odd->next=odd->next->next;
        even->next=even->next->next;
        even=even->next;
        odd=odd->next;
    }
    odd->next=evenHead;
    return head;
}

File: Brute_force.c
// 268. Missing Number
// Given an array nums containing n distinct numbers in the range [0, n], 
// return the only number in the range that is missing from the array.

  
int missingNumber(int* nums, int numsSize) {
    int n[numsSize+1];
    for(int i=0;i<=numsSize;i++){
        n[i]=0;
    }
    for(int i=0;i<numsSize;i++){
        n[nums[i]]=1;
    }
    for(int i=1;i<=numsSize;i++){
            if(n[i]!=1)
                return i;
    }
    return 0;
}
